stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -352,15 +316,11 @@ function mul!(C::AbstractMatrix, Da::Diagonal, Db::Diagonal, alpha::Number, beta
     return C
 end
 
+#TODO: many of /, \ related function has no size check and singular check
```

Comment:
I believe they are implicit in the `[l/r]div!` or in the broadcast...?

Reply:
./ and .\ doesn't do singular check. It returns nan directly.
So at least broadcast based r/div! misses that.

For size check, broadcast is enough for 2args div.
Although we can improve the error message.
For 3 arg ldiv, the output should have the same size in the 2nd dimension with  input.
Broadcast won’t throw error for cases like 10*10 out and 10*1 in.

Reply:
Yes, I was referring to the size checks.

Reply:
Checks?

Reply:
Why?

Reply:
If you mean size check, previous implement failed to throw with the following code:
```julia
ldiv!(randn(3,3), Diagonal(1:3), 1:3)
3×3 Matrix{Float64}:
 1.0  1.0  1.0
 1.0  1.0  1.0
 1.0  1.0  1.0
```

Reply:
Since the TODO has been removed, make this as resolved.